### PR TITLE
elb_application_lb: add support for creating alb with listener having multiple certificates

### LINKED ISSUE #1349: Multiple listener certificates for alb
### Summary

I am trying to add multiple certificates for a single load balancer, i.e. I'd like a load balancer that would be able to host sites that have the host name of foo.com and bar.com

In cloudformation, this can be done by using [AWS::ElasticLoadBalancingV2::ListenerCertificate](https://awstut.com/en/2022/02/19/attaching-multiple-acm-certificates-to-alb-to-support-multiple-domains-2/#toc9) and in boto3 this can be done with [add_listener_certificates()](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/elbv2.html#ElasticLoadBalancingv2.Client.add_listener_certificates)

### Issue Type

Feature Idea

### Component Name

amazon.aws.elb_application_lb_certificate

### Additional Information

```yaml
amazon.aws.elb_application_lb_certificate:
  listener: "{{ load_balancer.listeners[0].listener_arn }}"
  certificates:
    - CertificateArn: "{{ foo_certificate_arn }}"
      IsDefault: true
    - CertificateArn: "{{ bar_certificate_arn }}"
      IsDefault: false
```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by mandar242**
##### SUMMARY

Fixes #1349 
This PR adds support for adding multiple certificates to a listener during Application Load Balancer creation.

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

elb_application_lb

##### ADDITIONAL INFORMATION

As per boto3 documentation
1. only 1 certificate can be provided during call to [create_listener](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/elbv2/client/create_listener.html#create-listener)
2. Multiple certificates can be added using [add_listener_certificates](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/elbv2/client/add_listener_certificates.html), however important to note that only 1 certificate can be specified per call.

```paste below

```

### CODE REVIEW

File: changelogs/fragments/1950-elb_application_lb-multiple-listener-certificates-for-alb.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+---
+minor_changes:
+- elb_application_lb - Add support for adding listener with multiple certificates during ALB creation (https://github.com/ansible-collections/amazon.aws/pull/1950).
```
Comment (abikouo) [inline changelogs/fragments/1950-elb_application_lb-multiple-listener-certificates-for-alb.yml:3]: I would rather prefer stating that the change in on `module_utils/elbv2` and then mention that it brings benefit for the `elb_application_elb` module

File: plugins/module_utils/elbv2.py [hunk 2]
```diff
@@ -964,7 +964,26 @@ def add(self):
...
+                self.listener["Certificates"] = [first_certificate]
+                # create listener
+                create_listener_result = AWSRetry.jittered_backoff()(self.connection.create_listener)(
+                    LoadBalancerArn=self.elb_arn, **self.listener
+                )
+                # only one cert can be specified per call to add_listener_certificates
+                for cert in other_certs:
+                    AWSRetry.jittered_backoff()(self.connection.add_listener_certificates)(
+                        ListenerArn=create_listener_result["Listeners"][0]["ListenerArn"], Certificates=[cert]
+                    )
```
Comment (abikouo) [inline plugins/module_utils/elbv2.py:985]:
```suggestion
            listener_certificates = self.listener.get("Certificates", [])
            if len(listener_certificates) > 0:
                first_certificate, other_certs = listener_certificates[0], listener_certificates[1:]
                self.listener["Certificates"] = [first_certificate]
            # create listener
            create_listener_result = AWSRetry.jittered_backoff()(self.connection.create_listener)(
                LoadBalancerArn=self.elb_arn, **self.listener
            )
            # only one cert can be specified per call to add_listener_certificates
            for cert in other_certs:
                AWSRetry.jittered_backoff()(self.connection.add_listener_certificates)(
                    ListenerArn=create_listener_result["Listeners"][0]["ListenerArn"], Certificates=[cert]
                )
```

**Review by abikouo — CHANGES_REQUESTED** [review]
@mandar242 is it possible to update the corresponding integration test case?

**Review by abikouo — APPROVED** [review]
LGTM

**Review by alinabuzachis — APPROVED** [review]

**Comment by mandar242** [comment]
recheck

**Pull request merged**